toolchain/check/testdata/class/extern.carbon

```diff
@@ -0,0 +1,762 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+// ============================================================================
+// Setup files
+// ============================================================================
+
+// --- decl.carbon
+
+library "decl" api;
+
+class C;
+
+// --- extern_decl.carbon
+
+library "extern_decl" api;
+
+extern class C;
+
+// --- extern_decl_copy.carbon
+
+library "extern_decl_copy" api;
+
+extern class C;
+
+// --- def.carbon
+
+library "def" api;
+
+class C {}
+
+// ============================================================================
+// Test files
```

Comment:
Similar to how when you write a unit test you might have test infrastructure than multiple TEST_F files in a single file, we are moving towards a single file with multiple split-file test cases. The number of physical files is just becoming unwieldy, and copying setup will result in substantial duplication too.

Although the golden IR is significant, this is the trade-off we've chosen right now, putting the IR at the bottom of the file so that the test cases are easier to read through.